drivers/virtualbox/virtualbox.go

```diff
@@ -572,6 +573,44 @@ func (d *Driver) generateDiskImage(size int) error {
 	return createDiskImage(d.diskPath(), size, raw)
 }
 
+func (d *Driver) setupHostOnlyNetwork(machineName string) error {
+	ip, network, err := net.ParseCIDR(d.HostOnlyCIDR)
+	nAddr := network.IP.To4()
+
+	var dhcpAddr net.IP
+	// select pseudo-random DHCP addr; make sure not to clash with the host
+	for {
+		n := rand.Intn(25)
+		if byte(n) != nAddr[3] {
+			dhcpAddr = net.IPv4(nAddr[0], nAddr[1], nAddr[2], byte(1))
+			log.Debugf("using %s for dhcp address", dhcpAddr)
+			break
+		}
+	}
+
+	hostOnlyNetwork, err := getOrCreateHostOnlyNetwork(
+		ip,
+		network.Mask,
+		dhcpAddr,
+		net.IPv4(nAddr[0], nAddr[1], nAddr[2], byte(100)),
```

Comment:
Hm, just for clarity's sake, maybe we could assign these last two arguments to variables named `lowerBound` and `upperBound` and then pass those into this function?  It'd make it a bit easier to read, I think.